packages/babel-generator/src/generators/jsx.js

```diff
@@ -43,7 +44,8 @@ export function JSXSpreadChild(node: Object) {
 }
 
 export function JSXText(node: Object) {
-  this.token(node.value);
+  const raw = this.getPossibleRaw(node);
+  this.token(raw || node.value);
```

Comment:
> Can we explicitly check === null since raw being an empty string would also cause this to use .value instead? 

Yep, good catch – though I assume you mean `== null` ?

I'll respond to the second para in a follow-up comment, since this thread is about to disappear when I push.